### DISCUSSION TITLE
Resuming the bot after server/script restart

### DISCUSSION THREAD

**Discussion by doniyorbekm** [eternnoir/pyTelegramBotAPI#1941] [Q&A]
Hey everyone. I came across a problem when I made my bot and deployed it on a VPS server. Once I restart the script, my bot stops responding. I see that the webhook is set and Flask accepts requests, but my bot does not process any updates, thus acting like the script has not been run and **I have to send the /start command every time**. Here is my sample bot (before I moved it to Flask):
```
import telebot
from telebot import types

API_TOKEN = '61***74:AAGdCvKyW***k2n4jftR****n5eD_k'
bot = telebot.TeleBot(API_TOKEN)

user_dict = {}

class User:
    def __init__(self, name):
        self.name = name
        self.age = None
        self.sex = None


@bot.message_handler(commands=['help', 'start'])
def send_welcome(message):
    msg = bot.reply_to(message, """\
Hi there, I am Example bot.
What's your name?
""")
    bot.register_next_step_handler(msg, process_name_step)


def process_name_step(message):
    try:
        chat_id = message.chat.id
        name = message.text
        user = User(name)
        user_dict[chat_id] = user
        msg = bot.reply_to(message, 'How old are you?')
        bot.register_next_step_handler(msg, process_age_step)
    except Exception as e:
        bot.reply_to(message, 'oooops')


def process_age_step(message):
    try:
        chat_id = message.chat.id
        age = message.text
        if not age.isdigit():
            msg = bot.reply_to(message, 'Age should be a number. How old are you?')
            bot.register_next_step_handler(msg, process_age_step)
            return
        user = user_dict[chat_id]
        user.age = age
        markup = types.ReplyKeyboardMarkup(one_time_keyboard=True)
        markup.add('Male', 'Female')
        msg = bot.reply_to(message, 'What is your gender', reply_markup=markup)
        bot.register_next_step_handler(msg, process_sex_step)
    except Exception as e:
        bot.reply_to(message, 'oooops')


def process_sex_step(message):
    try:
        chat_id = message.chat.id
        sex = message.text
        user = user_dict[chat_id]
        if (sex == u'Male') or (sex == u'Female'):
            user.sex = sex
        else:
            raise Exception("Unknown sex")
        msg = bot.send_message(chat_id, 'Nice to meet you ' + user.name + '\n Age:' + str(user.age) + '\n Sex:' + user.sex)
        bot.register_next_step_handler(msg, process_name_step)
    except Exception as e:
        bot.reply_to(message, 'oooops')


bot.infinity_polling()
```
**So, my question is: how I modify this script so that my bot continues (resumes) working without having to send the /start command when my script is restarted?**

**Comment by coder2020official** [comment]
Hello. Can you provide more details? What are you getting when you try to use getWebhookInfo method(when bot does not work)? How are you restarting script, etc?

**Reply by doniyorbekm** [reply]
This is the response of getWebhookInfo at any time (even when the bot does not respond):
```
{
ok: true,
result: {
url: "https://domain.com:8443/hook/61***74:AAGdCv***pFk2n4jftR****5eD_k",
has_custom_certificate: false,
pending_update_count: 0,
max_connections: 40,
ip_address: "9******114"
}
}
```
I have this in my script (run with Flask):
```
app.run(host=WEBHOOK_LISTEN,
        port=WEBHOOK_PORT,
        ssl_context=(WEBHOOK_SSL_CERT, WEBHOOK_SSL_PRIV),
        debug=True)
```

and in the CLI:
1. I start the script with python app.py
2. My bot works
3. I stop the script with Ctrl+C
4. My bot does not work
5. I start the script with python app.py again
6. My bot still does not work, unless I send the /start command

**Reply by coder2020official** [reply]
getWebhookInfo response changes when you restart script. There should be an error shown if u try to send message to bot while it is disabled

**Reply by doniyorbekm** [reply]
Yes, you're right, getWebhookInfo() gives me a "Connection refused" error while my script is not running. But still, the problem persists. Once I restart my script, I have to send the /start command to make it work (which is bad user experience).

**Comment by coder2020official** [comment]
I'd recommend you to use built-in run_webhooks() method

**Reply by doniyorbekm** [reply]
Thanks, but I don't think this is a webhook issue. I have concerns about the register_next_step_handler() method.

**Reply by coder2020official** [reply]
Then don't use it. I prefer using states as an example. Register next step handler is not a reliable solution
